documentation/modules/oauth/con-oauth-authorization-keycloak-example.adoc

```diff
@@ -0,0 +1,452 @@
+Selecting it will open the `Realm Settings` for the `kafka-authz` realm.
+Next to the `Realm Settings` there are other sections we are interested in - `Groups`, `Roles`, `Clients` and `Users`.
+
+Under `Groups` we can see several groups that can be used to mark users as having some permissions.
+Groups are sets of users with name assigned. Typically they are used to geographically or organisationally compartmentalize users into organisations, organisational units, departments, and so on.
+
+In Keycloak the groups can be stored in an LDAP identity provider.
+That makes it possible to make some user a member of some group - through a custom LDAP server admin UI for example, which grants them some permissions on Kafka resources.
+
+Under `Users`, click on the `View all users` button and you will see two users defined - `alice` and `bob`. `alice` is a member of the `ClusterManager Group`, and `bob` is a member of `ClusterManager-my-cluster Group`.
+In Keycloak the users can be stored in an LDAP identity provider.
+
+Under `Roles` we can see several realm roles which can be used to mark users or clients as having some permissions.
+Roles are a concept analogous to groups. They are usually used to 'tag' users as playing organisational roles and having permissions that pertain to it.
+In Keycloak the roles cannot be stored in an LDAP identity provider - if that is your requirement then you should use groups instead.
+
+Under `Clients` we can see some additional clients configured - `kafka`, `kafka-cli`, `team-a-client`, `team-b-client`.
+The client with client id `kafka` is used by Kafka Brokers to perform the necessary OAuth2 communication for access token validation,
+and to authenticate to other Kafka Broker instances using OAuth2 client authentication.
+This client also contains Authorization Services resource definitions, policies and authorization scopes used to perform authorization on the Kafka Brokers.
+
+The client with client id `kafka-cli` is a public client that can be used by the Kafka command line tools when authenticating with username and password to obtain an access token or a refresh token.
+
+Clients `team-a-client`, and `team-b-client` are confidential clients representing services with partial access to certain Kafka topics.
+
+The authorization configuration is defined in the `kafka` client under `Authorization` tab.
+This tab becomes visible when `Authorization Enabled` is turned on under the `Settings` tab.
+
+
+## Authorization Services - Resources, Authorization Scopes, Policies and Permissions
+
+`Keycloak Authorization Services` use several concepts that together take part in defining, and applying access control to resources.
+
+`Resources` define _what_ we are protecting from unauthorized access.
+Each resource can contain a list of `authorization scopes` - actions that are available on the resource, so that permission on a resource can be granted for one or more actions only.
```

Comment:
```suggestion
Each resource can contain a list of authorization scopes - actions that are available on the resource, so that permission on a resource can be granted for one or more actions only.
```